WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/domainsfetcher/NewDomainsSearchRepository.kt

```diff
@@ -0,0 +1,72 @@
+package org.wordpress.android.ui.domains.management.newdomainsearch.domainsfetcher
+
+import org.wordpress.android.Constants
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.products.Product
+import org.wordpress.android.fluxc.store.ProductsStore
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnSuggestedDomains
+import org.wordpress.android.util.dispatchAndAwait
+import javax.inject.Inject
+
+private const val SUGGESTIONS_REQUEST_COUNT = 20
+
+class NewDomainsSearchRepository @Inject constructor(
```

Comment:
Btw, I'm working on unit tests for the repository 🧑‍🏭.  I've never tested EventBus-based code before so need a while to figure it out.

Reply:
Thank you for adding those test @justtwago 🙇
They look great 👍